### ISSUE TITLE
Exit Error Code on any error or warning.

### ISSUE THREAD

**Issue by Perkolator** [rhash/RHash#140]
Please implement a feature that would throw an error code on exit if ANY error/warning happens.

At the moment at least the "can't parse line" error/warning doesn't affect exit code.

If it's out of the question to make this a default behavior, then please implement this some other way. For example, a new option called `--strict`?

It would be nice to have this for detecting any errors with scripts (e.g. with [[ "$?" != "0" ]]).

Thank you!

(I wish that Linux Mint repo would pick up new RHash versions faster.. I'm new to linux and still learning how to get updated software)

**Comment by rhash** [comment]
For now I reject the feature in the form "throw an error code on exit if ANY error/warning", because it will take a lot of time to check all cases, and there are tasks with much higher priority. Please report errors/warnings with zero exit code, other than _"can't parse line"_, to handle them separately. Note, that all i/o errors already cause a non-zero exit code. 

According to the _"can't parse line"_ case, I agree, that this behavior should be changed to return a non-zero exit code. I think also some additional warning "Some lines could not be parsed" must be displayed at the end of a hash verification log. If the log file is very big or error is scrolled away, then the user should have an easy way to find the reason of the fail.

To simplify translators work and since broken hash files are pretty rare, the current error-reporting line _"Errors Occurred: Errors:%-3u Miss:%-3u Success:%-3u Total:%-3u"_ should be left unchanged. So the additional warning on an unparsed lines should be displayed on a separate line.

**Comment by rhash** [comment]
After some thought, I've end up with more simple solution. Now parsing errors are counted in statistics as "Errors", and thus affect the program exit code.

~~~sh
echo "abcd" | rhash -c -

--( Verifying (stdin) )---------------------------------------------------------
(stdin):1: can't parse line "abcd"
--------------------------------------------------------------------------------
Errors Occurred: Errors:1   Miss:0   Success:0   Total:1
~~~

IMO, adding a separate warning on unparsed lines could complicate parsing of RHash output by scripts or other software.

The feature is implemented as part of the 9105d546ea636b994337b71c976e1606306b1989 commit.
